api/src/extensions/manager.ts

```diff
@@ -307,10 +308,10 @@ export class ExtensionManager {
 
 		logger.info('Watching extensions for changes...');
 
-		const extensionDirUrl = pathToRelativeUrl(getExtensionsPath());
+		const extensionsDir = path.posix.resolve(getExtensionsPath());
 
-		const localExtensionUrls = NESTED_EXTENSION_TYPES.flatMap((type) => {
-			const typeDir = path.posix.join(extensionDirUrl, pluralize(type));
+		const nestedExtensions = NESTED_EXTENSION_TYPES.flatMap((type) => {
+			const typeDir = path.posix.join(extensionsDir, pluralize(type));
```

Comment:
This line is creating invalid paths resulting in legacy extension folders not being watched consistently on Windows.

![image](https://github.com/directus/directus/assets/9389634/e3aae01e-87d3-4345-a371-1c30afd5fe4f)

Reply:
<sub>Also in response to https://github.com/directus/directus/pull/21111/files#r1455398161</sub>

I've tested it on all platforms, including Windows.
It works on my end, the legacy extensions are watched too.
The paths are passed to `chokidar` in which they are correctly interpreted.

I went with `path.posix` on line 311 to be in accordance with `path.posix` on line 313 which was already there before. 
I'm not sure about the original reason for using `path.posix`, an assumption could be that there were concerns about globs with Windows-path-style.
I didn't want to change that as part of this PR, because as mentioned I don't know about the reason and it worked fine in my tests.

However, happy to update this if we can get some more clarity 👍

Reply:
The chokidar docs specifically mention that "globs must not contain windows separators (\), because that's how they work by the standard". That's why on line 311 `pathToRelativeUrl` was used, which converts any path to a relative URL (making it a relative path first ensures that we do not have to deal with windows vs posix root specifiers). So, even if it might works, maybe we should keep it that way to adhere to the standard 🤔

Reply:
Thank you @nickrum ❤️ 🙏 

The thing is that I need the absolute path to be able to apply https://github.com/directus/directus/pull/21111/files#diff-09fef1df59269b49b62b7e749e695089e0e51baa3f666521c50a48a58a7820f6R332

So in that case, could we use the following?
```js
const extensionsDir = pathToRelativeUrl(path.resolve(getExtensionsPath()));
```

Reply:
Alright! `pathToRelativeUrl` obviously won't work for absolute paths, so I gave it a try and removed all usages of `posix`. Turns out that works perfectly fine 👍🥳 
I think that is either because Node.js' `path` API already escapes the paths correctly or because `chokidar`'s statement is simply outdated.

The generated paths:
<img width="900" src="https://github.com/directus/directus/assets/5363448/9d5015fb-c701-4e12-9c70-6cba70278eb7">
 
Excerpt of chokidar's watched paths:
<img width="900" src="https://github.com/directus/directus/assets/5363448/ece410da-dc87-4c1d-9d08-35a1b8b66908">

Thanks @br41nslug thanks for checking on that, and thanks again @nickrum for the info! 🚀

Reply:
> The thing is that I need the absolute path to be able to apply https://github.com/directus/directus/pull/21111/files#diff-09fef1df59269b49b62b7e749e695089e0e51baa3f666521c50a48a58a7820f6R332

Do we actually need absolute paths for this to work, though? Couldn't this also be converted to a relative URL? Wouldn't that make sense given it is a glob pattern as well? I'm a little bit worried using something which is explicitly advised against. It might only work by chance and could break in a future minor/patch update 🤔

Reply:
Two things before I spend more time on this:
1. `chokidar` is more or less in unmaintained state, so I doubt there will be updates which break that behavior (or updates at all)
2. Over the long-term, I'd like to move away from `chokidar` anyway (see OP) because of 1) and the quirks it comes with

Reply:
Ah, I didn't even know that. Are there any good alternatives?

I really don't want to make your live any harder 🫣 So, if it works and as long as we keep in mind that this could break once we move away from chokidar, I'd be fine with it.

Reply:
> Ah, I didn't even know that. Are there any good alternatives?

Unfortunately, not really as of now 😢 There's [`parcel/watcher`](https://github.com/parcel-bundler/watcher) which looks promising but it's missing a high-level API, similar to `fs.watch`. I think https://github.com/vitejs/vite/issues/13593#issuecomment-1776906216 sums it up pretty well, for our needs too.

> I really don't want to make your live any harder 🫣

All good! ❤️ I just wanted to have that mentioned as basis for further discussion. I didn't mean to say that we can't make any further adjustments.

> So, if it works and as long as we keep in mind that this could break once we move away from chokidar, I'd be fine with it.

Yeah, that's what I'm also leaning towards at the moment.

Reply:
Seems to work fine, in my tests today 🤔 not sure if due to the changes or due to flawed testing in my original review.